server/src/main/java/keywhiz/service/daos/SecretSeriesMapper.java

```diff
@@ -20,26 +20,36 @@
 import com.fasterxml.jackson.databind.ObjectMapper;
 import java.io.IOException;
 import java.util.Map;
+import java.util.Optional;
 import javax.inject.Inject;
 import keywhiz.api.ApiDate;
+import keywhiz.api.model.Group;
 import keywhiz.api.model.SecretSeries;
 import keywhiz.jooq.tables.records.SecretsRecord;
+import keywhiz.service.config.Readonly;
 import org.jooq.RecordMapper;
 
 public class SecretSeriesMapper implements RecordMapper<SecretsRecord, SecretSeries> {
   private static final TypeReference<Map<String, String>> MAP_STRING_STRING_TYPE =
       new TypeReference<>() {};
+
   private final ObjectMapper mapper;
+  private final GroupDAO groupDAO;
 
-  @Inject public SecretSeriesMapper(ObjectMapper mapper) {
+  @Inject public SecretSeriesMapper(
+      ObjectMapper mapper,
+      @Readonly GroupDAO groupDAO) {
     this.mapper = mapper;
+    this.groupDAO = groupDAO;
   }
 
   public SecretSeries map(SecretsRecord r) {
+    String owner = getOwner(r);
```

Comment:
Similar to comment below, changing this to `String ownerName = getOwnerName(r)` makes it clearer that we are using `groupDAO` to find the owner name from the owner id.